09_variables_conditionals/solution_student/temp_converter/js/main.js

```diff
@@ -0,0 +1,45 @@
+
+
+$('#fahrenheit_to_celsius').on('click', function () {
+	$('body').removeClass('cold warm hot');
+	$('#result').removeClass('error-message');
+	$('#temperature').removeClass('error');
+	var inputVal = $('#temperature').val();
+	var resultVal = ((inputVal - 32) / 1.8);
+	var output = inputVal + '°F' + ' = ' + resultVal + '°C'
+	$('#result').html(output);
+
+
+	if (inputVal == "") {
```

Comment:
As you see, this works, because JavaScript is smart enough to recognize numbers in inputs, even if it is really a _string_, but it's usually best practice to convert strings, like text inputs, to floats/integers if you're going to work with them as numbers. So after you get your `inputVal`, you'd want to parseFloat it: `inputVal = parseFloat(inputVal);`. The only issue this will cause is in the conditional `if (inputVal == "")` — this will no longer work because the `inputVal` will either equal a number or `NaN` (not a number), so you would need to use `isNan(inputVal)` instead — so if it's a number then you can work with it, if it's not (which would include if it were just text), then give the error message. :)